src/lib/handlebars-converter/handlebars-helpers.js

```diff
@@ -32,7 +32,15 @@ var normalizeSectionName = function (name) {
 };
 
 var getDate = function (dateTimeString) {
+    // handling the date format here
     var ds = dateTimeString.toString();
+    if (/^(\d{0,5}|\d{7})$/.test(ds))
+        return ds;
+    if (/^\d{6}$/.test(ds))
+        return ds.substring(0, 4) + '-' + ds.substring(4, 6);
+    if (/^\d{8}$/.test(ds))
+        return ds.substring(0, 4) + '-' + ds.substring(4, 6) + '-' + ds.substring(6, 8);
+
```

Comment:
Is it possible to optimize the logic here? Can we first check for all digits and then check against length and/or move the common case to the top?

Reply:
OK. I changed the logic by checking the digits and then checking against length, like:
```
    if (!/^\d+$/.test(ds))
        return null;
    if (0 <= ds.length <= 5 && ds.length == 7)
        return ds;
    if (ds.length == 6)
        return ds.substring(0, 4) + '-' + ds.substring(4, 6);
    if (ds.length == 8)
        return ds.substring(0, 4) + '-' + ds.substring(4, 6) + '-' + ds.substring(6, 8);
```